lib/lambda_ethereum_consensus/state_transition/operations.ex

```diff
@@ -174,7 +174,56 @@ defmodule LambdaEthereumConsensus.StateTransition.Operations do
     Enum.reverse(withdrawals)
   end
 
-  @doc """
+  @spec process_attester_slashing(BeaconState.t(), SszTypes.AttesterSlashing.t()) ::
+          {:ok, BeaconState.t()} | {:ok, nil}
+  def process_attester_slashing(state, attester_slashing) do
+    attestation_1 = attester_slashing.attestation_1
+    attestation_2 = attester_slashing.attestation_2
+
+    cond do
+      Predicates.is_slashable_attestation_data(attestation_1.data, attestation_2.data) == false ->
+        {:ok, nil}
+
+      Predicates.is_valid_indexed_attestation(state, attestation_1) == false ->
+        {:ok, nil}
+
+      Predicates.is_valid_indexed_attestation(state, attestation_2) == false ->
+        {:ok, nil}
+
+      Predicates.is_indices_available(length(state.validators), attestation_1.attesting_indices) == false ->
+        {:ok, nil}
+
+      Predicates.is_indices_available(length(state.validators), attestation_2.attesting_indices) == false ->
+        {:ok, nil}
+
+      true ->
+        {slashed_any, state} =
+          Enum.uniq(attestation_1.attesting_indices)
+          |> Enum.filter(fn i -> Enum.member?(attestation_2.attesting_indices, i) end)
+          |> Enum.sort()
```

Comment:
Since they are already sorted, we don't need to sort them again

Reply:
This is not actually sorted in case you are mentioning the Enum.uniq() it just removes the duplicates
```elixir
iex(1)> a = [1, 4, 2, 3, 2, 4]
[1, 4, 2, 3, 2, 4]
iex(2)> Enum.uniq(a)
[1, 4, 2, 3]
```

Reply:
I thought `is_valid_indexed_attestation` checked that the list was sorted and had only unique values.

Reply:
you are actually right but 
https://github.com/lambdaclass/lambda_ethereum_consensus/blob/dd0a0523e6be2a21fa3f6a96d4466926307d236c/docs/specs/beacon-chain.md?plain=1#L2719C4-L2719C4
in here sort has been added again I don't know why

Reply:
😮 Let's leave it for now, then.